### PR TITLE
E UI 5449 extend organisation external controller

### PR DESCRIPTION
**Pull request by waynepHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-5449

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by tarikdogan-hmcts — APPROVED** [review]

File: api/pbas/index.ts [hunk 1]
```diff
@@ -1,64 +1,67 @@
...
+        const addPBAPath: string = `${url}/refdata/external/v1/organisations/pba`;
+        const pendingAddPBAs = paymentAccountRequest;
+        const addPBAPromise = req.http.post(addPBAPath, pendingAddPBAs);
+        allPromises.push(addPBAPromise);
         // do delete PBAs
-        const deletePBAPath: string = `${url}/api/pba/deletePBA`
-        const pendingRemovePBAs = pendingPaymentAccount.pendingRemovePaymentAccount
-        const deletePBAPromise = handleDelete(deletePBAPath, pendingRemovePBAs, req)
-        allPromises.push(deletePBAPromise)
+        // const deletePBAPath: string = `${url}/api/pba/deletePBA`
```
Comment (paulhowes-HMCTS) [inline api/pbas/index.ts:27]: Remove commented out code

File: config/default.json [hunk 2]
```diff
@@ -28,7 +28,6 @@
     "idamApi": "https://idam-api.platform.hmcts.net",
     "idamWeb": "https://hmcts-access.service.gov.uk",
     "rdProfessionalApi": "http://rd-professional-api-prod.service.core-compute-prod.internal",
-    "s2s": "http://rpe-service-auth-provider-prod.service.core-compute-prod.internal",
```
Comment (paulhowes-HMCTS) [inline config/default.json:31]: Query - why have you removed the s2s config setting?

File: src/organisation/components/pba-numbers-form/pba-numbers-form.component.spec.ts [hunk 3]
```diff
@@ -6,20 +6,17 @@ import { RouterTestingModule } from '@angular/router/testing';
 import { ExuiCommonLibModule } from '@hmcts/rpx-xui-common-lib';
 import { Action, Store } from '@ngrx/store';
 import { RxReactiveFormsModule } from '@rxweb/reactive-form-validators';
-import { of } from 'rxjs';
-
+import { of } from 'rxjs/internal/observable/of';
```
Comment (paulhowes-HMCTS) [inline src/organisation/components/pba-numbers-form/pba-numbers-form.component.spec.ts:9]: I don't think this should be deep nested - suggest reverting back to import { of } from 'rxjs';

File: src/organisation/containers/organisation/organisation.component.html [hunk 4]
```diff
@@ -26,16 +24,18 @@
...
       <td class="govuk-table__cell">
         <a class="govuk-link fright" routerLink="/organisation/update-pba-numbers" *ngIf="showChangePbaNumberLink" id="change-pba-account-numbers__link" aria-label="Change PBA account numbers">Change</a>
-
-        <div *ngIf="organisationPaymentAccount">
-          <div *ngFor="let pba of organisationPaymentAccount">{{pbaNumberWithStatus(pba)}}</div>
-        </div>
-
-        <div *ngIf="!organisationPaymentAccount">
-          <div>No PBA accounts registered</div>
+          <div *ngIf="organisationPaymentAccount">
```
Comment (paulhowes-HMCTS) [inline src/organisation/containers/organisation/organisation.component.html:30]: I think there's too much nesting here. Doesn't need the outer if statement as the ngFor will take care of the visibility.

File: src/organisation/containers/organisation/organisation.component.html [hunk 5]
```diff
@@ -26,16 +24,18 @@
...
+            <div *ngFor="let pba of organisationPaymentAccount">{{pba.pbaNumber}} (Accepted)</div>
+          </div>
+          <div *ngIf="!organisationPaymentAccount">
```
Comment (paulhowes-HMCTS) [inline src/organisation/containers/organisation/organisation.component.html:33]: Don't need the nested div which is inside the ngIf

File: src/organisation/containers/organisation/organisation.component.html [hunk 6]
```diff
@@ -26,16 +24,18 @@
...
+            <div>No PBA accounts registered</div>
+          </div>
+          <div *ngFor="let pendingPBA of organisationPendingPaymentAccount">{{pendingPBA}}  (Pending)</div>
+          <div *ngIf="!organisationPendingPaymentAccount">
+          <div>No Pending PBA accounts registered</div>
```
Comment (paulhowes-HMCTS) [inline src/organisation/containers/organisation/organisation.component.html:38]: Don't need the <div> around No Pending PBA accounts registered as it is contained within the ngIf div

**Review by paulhowes-HMCTS — CHANGES_REQUESTED** [review]
A few minor changes and a query. Please make necessary changes, reply to my comments and pass back to me so I can resolve my comments if ok.

File: api/pbas/index.ts (hunk 1 above)
Comment (waynepHMCTS) [reply api/pbas/index.ts]: removed

File: config/default.json (hunk 2 above)
Comment (waynepHMCTS) [reply config/default.json]: Put it back in

File: src/organisation/components/pba-numbers-form/pba-numbers-form.component.spec.ts (hunk 3 above)
Comment (waynepHMCTS) [reply src/organisation/components/pba-numbers-form/pba-numbers-form.component.spec.ts]: Its now shallow nested

File: src/organisation/containers/organisation/organisation.component.html (hunk 4 above)
Comment (waynepHMCTS) [reply src/organisation/containers/organisation/organisation.component.html]: Remove unnecessary div

File: src/organisation/containers/organisation/organisation.component.html (hunk 5 above)
Comment (waynepHMCTS) [reply src/organisation/containers/organisation/organisation.component.html]: I agree removed external div

File: src/organisation/containers/organisation/organisation.component.html (hunk 6 above)
Comment (waynepHMCTS) [reply src/organisation/containers/organisation/organisation.component.html]: I removed unnecessary external div

File: src/organisation/containers/organisation/organisation.component.html (hunk 4 above)
Comment (paulhowes-HMCTS) [reply src/organisation/containers/organisation/organisation.component.html]: Formatting of the div is a little out but I'll resolve for now.

File: src/organisation/containers/organisation/organisation.component.html (hunk 5 above)
Comment (paulhowes-HMCTS) [reply src/organisation/containers/organisation/organisation.component.html]: Resolved.

File: src/organisation/containers/organisation/organisation.component.html (hunk 6 above)
Comment (paulhowes-HMCTS) [reply src/organisation/containers/organisation/organisation.component.html]: Resolved.

File: api/pbas/index.ts (hunk 1 above)
Comment (paulhowes-HMCTS) [reply api/pbas/index.ts]: Resolved.

**Review by paulhowes-HMCTS — APPROVED** [review]
Approved after changes.

**Pull request closed without merging**